ibis/expr/rules.py

```diff
@@ -77,6 +77,7 @@ def shape_like(name):
     def shape(self):
         args = getattr(self, name)
         args = args if util.is_iterable(args) else [args]
+        args = [a for a in args if a is not None]
```

Comment:
Can we handle this in `highest_precedence_shape`?

Reply:
I was taking your cue from https://github.com/ibis-project/ibis/pull/9333#issuecomment-2155523018 and trying to only push this logic as far down as it needs to go. For example, Operations with optional arguments can still use `shape_like()` and this deals with that case, so those ops don't need to add in their own filtering logic. I thought that was common enough between Operations that it was worth factoring to here. But, other places in ibis besides Values might want to use `highest_precedence_shape`, and I think they should have to deal with `None`s in their own way. If one of those users doesn't deal with Nones, we want that to raise an error, not silently give the wrong answer.

This logic was also why I didn't add `args = [a for a in args if a is not None]` to `dtype_like()` above: It isn't needed yet, and if a user of dtyp_like ever starts accidentally passing in Nones, I want there to be an error so that we can notice and then decide what to do.

I can push this down to `highest_precedence_shape` though if you still think that's a good idea.

Reply:
Ironically, now that you are making me think of it, I actually now want to push the `args = args if util.is_iterable(args) else [args]` logic down into `highest_precedence_shape`, because I think that would NEVER be a footgun. What do you think, should I do that (maybe in a separate PR)?

Reply:
oh, also blocking for #9096

@cpcloud I am fine with pushing logic down a level further, just let me know what you think

Reply:
This is fine as is for now! Thanks!